tests/jballerina-unit-test/src/test/resources/test-src/expressions/typecast/type-casting.bal

```diff
@@ -977,6 +977,32 @@ function testCastOfTwoDimensionalIntArrayToByteArray() {
     assertEquality("[[1,2,3],[4,5,6]]", c.toString());
 }
 
+function testCastJsonToMapOfAnydata() {
+    json j1 = {a1: (), a2: 1, a3: 1.0f, a4: 1.2d, a5: "a5", a6: true};
+    map<anydata> m1 = <map<anydata>> j1;
+
+    json j2 = {a1: null, a2: 1, a3: 1.0, a4: 1d, a5: "a5", a6: false, a7: {a1: j1}};
+    map<anydata> m2 = <map<anydata>> j2;
+
+    assertEquality("{\"a1\":null,\"a2\":1,\"a3\":1.0,\"a4\":1.2,\"a5\":\"a5\",\"a6\":true}", m1.toString());
+    assertEquality("{\"a1\":null,\"a2\":1,\"a3\":1.0,\"a4\":1,\"a5\":\"a5\",\"a6\":false,\"a7\":{\"a1\":{\"a1\":null,\"a2\":1,\"a3\":1.0,\"a4\":1.2,\"a5\":\"a5\",\"a6\":true}}}", m2.toString());
```

Comment:
When we cast to `map<anydata>`, shouldnt `null` be changed to `()`?

Reply:
Yeah. It is most likely a bug.

Reply:
Shall I create a separate issue for this or can we fix it in this PR?

Reply:
> When we cast to `map<anydata>`, shouldnt `null` be changed to `()`?

In both cases the value is a single value (Java `null` in jBallerina). In Ballerina `()` vs `null` is just two ways of writing the same value. Generally, at runtime, you wouldn't know how it was specified. 

> Yeah. It is most likely a bug because spec says `the use of null should be restricted to JSON-related contexts`.

This is just a recommendation for users. https://github.com/ballerina-platform/ballerina-lang/issues/28342#issuecomment-806403455

We are calling `value:toString()` here. So the relevant sections in the spec are
- https://github.com/ballerina-platform/ballerina-spec/blob/master/lang/lib/value.bal#L78
- https://ballerina.io/spec/lang/master/#ToString

Based on these, do you think it is a bug?

Reply:
No. Thanks for the explanation.